core/capabilities/remote/target/request/client_request.go

```diff
@@ -56,9 +58,14 @@ func NewClientRequest(ctx context.Context, lggr logger.Logger, req commoncap.Cap
 	lggr.Debugw("sending request to peers", "execID", req.Metadata.WorkflowExecutionID, "schedule", peerIDToTransmissionDelay)
 
 	responseReceived := make(map[p2ptypes.PeerID]bool)
+
+	ctx, cancelFn := context.WithCancel(context.Background())
```

Comment:
It could still be valuable/important to connect an upstream context from the method args. If we use `context.WithoutCancel`, then the async work will continue in the background after the parent context is done, and we'll still inherit all of the request scoped key/vals associated with that parent.
```suggestion
	ctx, cancelFn := context.WithCancel(context.WithoutCancel(ctx))
```
Perhaps there is a more clear way to organize & document what is happening though :thinking:

Reply:
yeah, thats better.  Updated.